airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java

```diff
@@ -280,4 +273,21 @@ private Optional<StandardWorkspace> getWorkspaceIfExists(final UUID workspaceId,
     }
   }
 
+  /**
+   * No frills, given a coordinate, just store the payload
+   */
+  public SecretCoordinate storeOAuthSecret(final UUID workspaceId, final String payload) {
```

Comment:
Ideally this class, which is a generic utility for writing secrets used by all of the platform (whereas OAuth feels more like a specific application feature that shouldn't proliferate across horizontal utilities), doesn't have any special conception or knowledge of OAuth. Could we instead inject the coordinate base prefix from the calling context, since that does have knowledge of OAuth?

Reply:
Definitely we could. Will move that to the calling context.